rewrite-java-core/src/test/java/BinaryDataTest.java

```diff
@@ -0,0 +1,42 @@
+import org.intellij.lang.annotations.Language;
+import org.junit.jupiter.api.Test;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+public class BinaryDataTest implements RewriteTest {
+    /**
+     * BinaryDataTest tests the recipe that changes
+     * com.azure.core.util.BinaryData to io.clientcore.core.util.binarydata.BinaryData.
+     * @author Ali Soltanian Fard Jahromi
+     */
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipeFromResource("/META-INF/rewrite/rewrite.yml",
+                "com.azure.rewrite.java.core.MigrateAzureCoreSamplesToAzureCoreV2");
+    }
+
+    /* Test to make sure BinaryData type and import is changed */
+    @Test
+    public void testClientLoggerTypeAndImportChanged() {
```

Comment:
We should also add tests that use TypeReference in BinaryData.

Reply:
Added unit test that tests BinaryData change with TypeReference in commit [5447e27](https://github.com/Azure/azsdk-java-rewrite-recipes/pull/14/commits/5447e27b96a32983846088fe796effd5c4256353)